### PR TITLE
reduce usage of forced unwarps

### PR DESCRIPTION
**Pull request by tomerd**
motivation: less crashes, happier users

changes: replace usage of forced unwraps with throwing guards, use InternalError for assertion during developement

### CODE REVIEW

**Comment by tomerd** [comment]
@swift-ci please smoke test

File: Sources/Build/BuildPlan.swift [hunk 1]
```diff
@@ -1782,7 +1794,7 @@ public class BuildPlan {
 
         pkgConfigCache[target] = ([String](cflagsCache), libsCache)
 
-        return pkgConfigCache[target]!
+        return pkgConfigCache[target]! // forced unwrap safe
```
Comment (abertelrud) [inline Sources/Build/BuildPlan.swift:1797]: For these I usually tend to do the more verbose thing of creating it to a local variable, then storing it in the dictionary, and then returning the local variable.  It's more verbose but avoids having to explain a force unwrap.  But it's of course a matter of preference.

**Review by abertelrud — APPROVED** [review]

File: Sources/Build/BuildPlan.swift (hunk 1 above)
Comment (tomerd) [reply Sources/Build/BuildPlan.swift:1797]: > For these I usually tend to do the more verbose thing of creating it to a local variable, then storing it in the dictionary, and then returning the local variable

not sure I follow your technique, can you give an example?

Comment (tomerd) [reply Sources/Build/BuildPlan.swift:1797]: oh I see... before setting it in the "cache" you create a local variable. makes sense

Comment (abertelrud) [reply Sources/Build/BuildPlan.swift:1797]: Yeah, I just really dislike force unwraps...  so am willing to have more verbiage.  But that's probably just a question of style.

Comment (abertelrud) [reply Sources/Build/BuildPlan.swift:1797]:
```
let foo = bar()
baz[something] = foo
return foo
`

Comment (tomerd) [reply Sources/Build/BuildPlan.swift]: add a fixup to adopt this. ptal

**Comment by tomerd** [comment]
@swift-ci please smoke test

File: Sources/Build/BuildPlan.swift (hunk 1 above)
Comment (abertelrud) [reply Sources/Build/BuildPlan.swift]: Looks great.  Thanks!

**Pull request merged**